src/node/rpc/call_types.h

```diff
@@ -65,38 +55,42 @@ namespace ccf
 
   struct GetNetworkInfo
   {
-    struct NodeInfo
-    {
-      NodeId node_id;
-      std::string host;
-      std::string port;
-    };
-
     struct Out
     {
-      std::vector<NodeInfo> nodes = {};
-      std::optional<NodeId> primary_id = std::nullopt;
+      ServiceStatus service_status;
+      std::optional<kv::Consensus::View> current_view;
+      std::optional<NodeId> primary_id;
     };
   };
 
-  struct GetNodesByRPCAddress
+  struct GetNode
   {
     struct NodeInfo
     {
       NodeId node_id;
       NodeStatus status;
+      std::string host;
```

Comment:
Why not use `NodeInfoNetwork` here directly? Some fields (e.g. public interface) are missing for example.

Reply:
Ah, checked the implementation of the endpoints that populate this struct and we do return the public port. In fact, we'd report the public and private RPC interfaces. Is this expected? If my understand is correct, one seems interesting for operators (currently `rpc_host` and `rpc_port`) while the public interface is mostly for users/members?

Reply:
I used a new type to provide a boundary between internal and public interface, since we don't want to break the latter even if the internal one changes. 
And yes, returning the public and private addresses is expected, this was an ask from @lynshi and I don't see any harm in it either way.
We should discuss naming, rpc_* doesn't seem like a great choice.

Reply:
These could be exposed as `private_host/port` perhaps?

Reply:
`local` would preferable to `private`, these are the local interfaces the nodes were told to bind to on startup. Whether they differ from `public` depends on whether there is some routing/NATing going on, but they aren't really `private`, at least not in way that compares to other things being private in CCF.

Reply:
What about `nodehost`/`nodeport`? Does it make sense to expose that as well?

Reply:
@letmaik These are only used for node-to-node communications and aren't exposed to clients directly so my hunch is that they shouldn't be exposed directly here.

Reply:
OK, I renamed it to `local_*`